src/lib/datepicker/calendar.ts

```diff
@@ -207,6 +205,22 @@ export class MdCalendar<D> implements AfterContentInit {
     }
   }
 
+  /** Focuses the active cell after the microtask queue is empty. */
+  _focusActiveCell() {
+    this._ngZone.onMicrotaskEmpty.first().subscribe(() => {
```

Comment:
This is a good place for `runOutsideAngular`, since otherwise the running of this microtask will cause another round of change detection
```ts
this._ngZone.runOutsideAngular(() => {
  Promise.resolve().then(() => {
    // ...
  });
});
```